qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -364,42 +275,83 @@ def to_list(self) -> List[Tuple[str, complex]]:
         Returns:
             A list of tuples consisting of the dense label and corresponding coefficient.
         """
-        return list(zip(self._labels, self._coeffs.tolist()))
+        if FermionicOp._label_display_mode_is_dense:
+            return self._to_legacy().to_list()
+        return self._data.copy()
 
     def adjoint(self) -> "FermionicOp":
-        dagger_map = {"+": "-", "-": "+", "I": "I", "N": "N", "E": "E"}
-        label_list = []
-        coeff_list = []
-        for label, coeff in zip(self._labels, self._coeffs.tolist()):
+        data = []
+        for label, coeff in self._data:
             conjugated_coeff = coeff.conjugate()
+            adjoint_label = " ".join(
+                "+" + c[1:] if c[0] == "-" else "-" + c[1:] for c in reversed(label.split())
+            )
+            data.append((adjoint_label, conjugated_coeff))
 
-            daggered_label = []
-            count = label.count("+") + label.count("-")
-            for char in label:
-                daggered_label.append(dagger_map[char])
-                if char in "+-":
-                    count -= 1
-                    if count % 2 == 1:
-                        conjugated_coeff *= -1
-
-            label_list.append("".join(daggered_label))
-            coeff_list.append(conjugated_coeff)
-
-        return FermionicOp(list(zip(label_list, np.array(coeff_list, dtype=np.complex128))))
+        return FermionicOp(data, register_length=self.register_length)
 
     def reduce(self, atol: Optional[float] = None, rtol: Optional[float] = None) -> "FermionicOp":
         if atol is None:
             atol = self.atol
         if rtol is None:
             rtol = self.rtol
 
-        label_list, indices = np.unique(self._labels, return_inverse=True, axis=0)
-        coeff_list = np.zeros(len(self._coeffs), dtype=np.complex128)
-        for i, val in zip(indices, self._coeffs):
-            coeff_list[i] += val
-        non_zero = [
-            i for i, v in enumerate(coeff_list) if not np.isclose(v, 0, atol=atol, rtol=rtol)
-        ]
-        if not non_zero:
-            return FermionicOp(("I" * self.register_length, 0))
-        return FermionicOp(list(zip(label_list[non_zero].tolist(), coeff_list[non_zero])))
+        op = self._to_legacy().reduce(atol, rtol)
+        return FermionicOp._from_legacy(op)
+
+    @classmethod
+    def set_label_display_mode(cls, mode: str):
+        """Set the display mode of labels.
+
+        Args:
+            mode: display mode of labels. "sparse" or "dense" is available.
+
+        Raises:
+            ValueError: invalid mode is given
+        """
+        mode_lower = mode.lower()
+        if mode_lower == "dense":
+            cls._label_display_mode_is_dense = True
+        elif mode_lower == "sparse":
+            cls._label_display_mode_is_dense = False
+        else:
+            raise ValueError(f"Invalid `mode` {mode} is given. `mode` must be dense or sparse.")
```

Comment:
```suggestion
            raise ValueError(f"Invalid `mode` {mode} is given. `mode` must be 'dense' or 'sparse'.")
```
Just to make it that tiny bit more explicit what is meant here.